icu4j/main/classes/core/src/com/ibm/icu/impl/locale/LocaleDistance.java

```diff
@@ -340,13 +342,29 @@ public int getBestIndexAndDistance(LSR desired, LSR[] supportedLSRs,
                 // Distinguish between equivalent but originally unequal locales via an
                 // additional micro distance.
                 shiftedDistance |= (desired.flags ^ supported.flags);
-            }
-            if (shiftedDistance < shiftedThreshold) {
-                if (shiftedDistance == 0) {
-                    return slIndex << INDEX_SHIFT;
+                if (shiftedDistance < shiftedThreshold) {
+                    if (shiftedDistance == 0) {
+                        return slIndex << INDEX_SHIFT;
+                    }
+                    bestIndex = slIndex;
+                    shiftedThreshold = shiftedDistance;
+                    bestLikelyInfo = -1;
+                }
+            } else {
+                if (shiftedDistance < shiftedThreshold) {
```

Comment:
My eyes might be bleary, but it looks like this test is the same as line 345.

Reply:
Right. I duplicated it and made it do slightly different things for equivalent desired&supported vs. not equivalent. Otherwise I would have to add another variable to remember equivalence, or see if the distance is at least 1.0 and rely on the distance shift to be large enough to keep the lsr flags smaller than 1.0.